File: cpp/Array/L_countNicePairInArr.cpp
#include <bits/stdc++.h>
using namespace std;

int countNicePairs(vector<int> &nums)
{
    int res = 0;
    const int mod = 1e9 + 7;
    unordered_map<int, int> count;

    for (int n : nums)
    {
        int r = rev(n);
        int cur = count[n - r];
        count[n - r] = cur + 1;
        res = (res + cur) % mod;
    }

    return res;
}

int rev(int num)
{
    int reversedNum = 0;

    while (num > 0)
    {
        int digit = num % 10;
        reversedNum = reversedNum * 10 + digit;
        num = num / 10;
    }

    return reversedNum;
}

File: Python/Arrays/L_countNicePairInArr.py
from typing import List


def countNicePairs(nums: List[int]) -> int:
    res = 0
    count = {}
    mod = 10**9 + 7
    
    for n in nums:
        rev = int(str(n)[::-1])
        cur = count.get(n - rev, 0)
        res += cur
        count[n - rev] = 1 + cur

    return res % mod